base/dict.jl

```diff
@@ -485,18 +477,16 @@ function get!(default::Callable, h::Dict{K,V}, key::K) where V where K
     end
     if index > 0
         h.age += 1
-        @inbounds h.keys[index] = key
-        @inbounds h.vals[index] = v
+        @inbounds h.pairs[index] = Pair{K,V}(key, v)
     else
         @inbounds _setindex!(h, v, key, -index)
     end
     return v
 end
 
-
 function getindex(h::Dict{K,V}, key) where V where K
     index = ht_keyindex(h, key)
-    @inbounds return (index < 0) ? throw(KeyError(key)) : h.vals[index]::V
+    @inbounds return (index < 0) ? throw(KeyError(key)) : h.pairs[index].second::V
```

Comment:
I wonder why all these type annotations here were added in the first place. They probably don't hurt, but I also don't see why they'd be needed.